js/server/modules/@arangodb/foxx/queues/manager.js

```diff
@@ -128,26 +139,47 @@ const resetDeadJobs = function () {
 
   const initialDatabase = db._name();
   db._databases().forEach(function (name) {
-    try {
-      db._useDatabase(name);
-      var ok = false;
-      while (!ok) {
-        try {
-          db._query(query);
-          ok = true;
-        } catch(e) {
-          warn("Exception while resetting dead jobs " + e.message, " retrying in 10s");
+    var done = false;
+    // The below code retries under the assumption that it should be
+    // sufficient that
+    //   * the database exists
+    //   * the collection _jobs exists
+    //   * we are not shutting down
+    // for this operation to eventually succeed work.
+    // If any one of the above conditions is violated we abort,
+    // otherwise we retry for some time (currently a hard-coded minute)
+    var maxTries = 6;
+    while (!done && maxTries > 0) {
+      try {
+        // this will throw when DB does not exist (anymore)
+        db._useDatabase(name);
+
+        // this might throw if the _jobs collection does not
+        // exist (or the database was deleted between the
+        // statement above and now...)
+        db._query(query);
+
+        // Now the jobs are reset
+        if (!isCluster) {
+          queues._updateQueueDelay();
+        } else {
+          global.KEYSPACE_CREATE('queue-control', 1, true);
+        }
+        done = true;
+      } catch(e) {
+        if (e.code === errors.ERROR_SHUTTING_DOWN.code) {
+          warn("Shutting down while resetting dead jobs on database " + name + ", aborting.");
+          done = true; // we're quitting because shutdown is in progress
+        } else if (e.code === errors.ERROR_ARANGO_DATA_SOURCE_NOT_FOUND.code) {
+          warn("Data source not found while resetting dead jobs on database " + name + ", aborting.");
+          done = true; // we're quitting because shutdown is in progress
```

Comment:
```suggestion
          done = true; // we're quitting because the collection is missing, most probably the database has been deleted meanwhile.
```